packages/web5-user-agent/src/web5-user-agent.ts

```diff
@@ -280,6 +322,54 @@ export class Web5UserAgent implements Web5Agent {
     return { message: dwnMessage.toJSON(), dataStream: readableStream };
   }
 
+
+  // TODO: have issuer did key already stored in key manager and use that instead of generating a new one
```

Comment:
You got it -- you'd want to reference the key that's already stored.

Worth noting that there's a convention we've been using for signing DWN messages and encrypting DWN records, which is noted here:
- https://github.com/TBD54566975/web5-js/blob/d8e2be7380e1ad504e7467d990dd942cead04ee5/packages/dids/src/types.ts#L61-L66
- https://github.com/TBD54566975/web5-js/blob/d8e2be7380e1ad504e7467d990dd942cead04ee5/packages/dids/src/did-ion.ts#L175-L182

To simplify we're going to combine `messageAuthorizationKeys` and `messageAttestationKeys`  to `messageSigningKeys` since the same key can be used for both.  Attestation is an option for a message if the author wants non-repudiable attestation.  For now, you can resolve the DID document of the other, get the service endpoints using [`getServices()`](https://github.com/TBD54566975/web5-js/blob/d8e2be7380e1ad504e7467d990dd942cead04ee5/packages/dids/src/utils.ts#L47-L53) from the `dids` package and use the first index in the `messageAuthorizationKeys` array as the key reference to ask the `KeyManager` to sign with.